CommonTools/TrackerMap/src/TrackerMap.cc

```diff
@@ -3611,3 +3619,37 @@ for (int layer=1; layer < 44; layer++){
   }
 saveAsSingleLayer=false;
 }
+
+TLegend* TrackerMap::buildLegend() {
+  if ( legInfos_.empty() ) {
+    legInfos_.reserve(8);
+    legInfos_.push_back(new TH1F("green","green",1,0.,1.));
```

Comment:
TPolyLine looks like a much cheaper alternative to the full histogram
```
legInfos_.push_back(new TPolyLine(0));
```

Reply:
Status note: I'm waiting for this to be resolved

Reply:
You are really shooting a fish in a barrel with this too long standing pull request: the TrackerMap code should not be improved and, instead, the implementation of the needed features in the TH2Poly should be encouraged by the CMS software and offline group. In this way the Tracker group can abandon this obsolete piece of code.

Reply:
Attempt of this with the DQM group : https://hypernews.cern.ch/HyperNews/CMS/get/dqmDevel/2394.html

Reply:
I see: in that thread we are encouraged NOT to use root stuff to do that job because it is too much resource consuming while in this pull request we are strongly pushed to use root to add a legend to something which is not root based (because of the problem above).

Reply:
This is the only file and modifications in the reconstruction category.
My review was purely incremental.
IIRC one empty TH1F is about 1kB. An empty TPolyLine should be much less than that.
If the better solution is to close this PR and find a different solution, I would not mind it at all.

Reply:
@fioriNTU I am still missing what prevents us (Tracker) from abandoning completely the TrackerMap of Pino Zito and use the ones which are provided by tkCommissioner for ALL the maps which are produced by the offline shift crew. This is regardless if we ever have TH2Poly and TProfilePoly in the DQM GUI or in CMSSW to fill them during a possible harvesting step (as we are doing now with Pino Zito's maps).

Reply:
@venturia in principle we may use the TkCommissioner to check all the TkMaps, the problem is that in that case ou have to run an application (similar to a root TBrowser), that is not much confortable when in remote. 
However, @slava77 and @davidlange6 I think this should be merged at this point ... what is really missing?

Reply:
@fioriNTU we can easily print in png files the maps produced by tkCommissioner instead of printing the one produced by TrackerMap code, can't we?

Reply:
This is for sure possible somehow, I will check the details and come back to you.